### PR TITLE
Coverity fixes

### PR DESCRIPTION
**Pull request by cabelitos**

### CODE REVIEW

File: src/modules/flow/persistence/persistence.c [hunk 1]
```diff
@@ -676,10 +676,10 @@ persist_string_open(struct sol_flow_node *node,
...
         SOL_FLOW_NODE_TYPE_PERSISTENCE_STRING_OPTIONS_API_VERSION, -EINVAL);
 
-    if (opts->default_value) {
-        mdata->default_value = strdup((char *)opts->default_value);
-        SOL_NULL_CHECK(mdata->default_value, -ENOMEM);
-    }
+    SOL_NULL_CHECK_MSG(opts->default_value, -EINVAL,
+        "The default value can't be NULL");
+    mdata->default_value = strdup(opts->default_value);
+    SOL_NULL_CHECK(mdata->default_value, -ENOMEM);
```
Comment (edersondisouza) [inline src/modules/flow/persistence/persistence.c]: OK. But as the only way to get `NULL` here is by it being set to `persistence.json` - which is ours - maybe could you simplify and just get rid of all null checks? Or is this too outrageous?

**Comment by bdilly** [comment]
@edersondisouza is right

It's the way I would prefer to go

File: src/modules/flow/persistence/persistence.c (hunk 1 above)
Comment (cabelitos) [inline src/modules/flow/persistence/persistence.c]: Indeed.

**Comment by cabelitos** [comment]
@edersondisouza @bdilly Sent a v2. I got rid of the null checks

**Comment by edersondisouza** [comment]
r+

**Comment by cabelitos** [comment]
merged

**Pull request closed without merging**